internal/beater/beater.go

```diff
@@ -180,36 +180,47 @@ func (s *Runner) Run(ctx context.Context) error {
 
 	// Obtain the memory limit for the APM Server process. Certain config
 	// values will be sized according to the maximum memory set for the server.
-	var memLimit float64
+	var memLimitGB float64
 	if cgroupReader := newCgroupReader(); cgroupReader != nil {
 		if limit, err := cgroupMemoryLimit(cgroupReader); err != nil {
 			s.logger.Warn(err)
 		} else {
-			// Limit the memory to 80% of the cgroup memory limit.
-			memLimit = float64(limit) / 1024 / 1024 / 1024 * 0.8
+			memLimitGB = float64(limit) / 1024 / 1024 / 1024
 		}
 	}
-	if memLimit <= 0 {
+	if memLimitGB <= 0 {
 		s.logger.Info("no cgroups detected, falling back to total system memory")
 		if limit, err := systemMemoryLimit(); err != nil {
 			s.logger.Warn(err)
 		} else {
-			// If no cgroup limit is set, only return 50% of the total memory.
+			// If no cgroup limit is set, return a fraction of the total memory
 			// to have a margin of safety for other processes.
-			memLimit = float64(limit) / 1024 / 1024 / 1024 * 0.5
+			memLimitGB = float64(limit) / 1024 / 1024 / 1024 * 0.625
```

Comment:
@marclop I have updated the code to give us the actual memory limit in GB (previously we were using 80%). To keep the logic for setting max concurrent decoders same I have moved the 80% requirement inside the max decoder scope.

For the highlighted section I have updated the memory limit to be 62.5% of max so that 80% of this value comes out to be 50% of total. Let me know if you have any concerns.